testhelpers/tar_assertions.go

```diff
@@ -113,6 +136,19 @@ func SymlinksTo(expectedTarget string) TarEntryAssertion {
 	}
 }
 
+func HardLinks() TarEntriesAssertion {
+	return func(t *testing.T, header1 *tar.Header, _ []byte, header2 *tar.Header, _ []byte) {
+		t.Helper()
+		if header1.Typeflag != tar.TypeLink && header2.Typeflag != tar.TypeLink {
```

Comment:
I'm slightly confused - this code updates the header type only for 2nd (and later) instances of the file:

```
		if processedPath, ok := hardLinkFiles[inode]; ok {
			header.Typeflag = tar.TypeLink
			header.Linkname = processedPath
			header.Size = 0
		} else {
			hardLinkFiles[inode] = header.Name
		}
```

How do both headers get type flag `tar.TypeLink`?

Reply:
Actually I am checking that at least one of them is a hardlink, but I don't know which one is going to be pick up first